timezones.nim

```diff
@@ -158,14 +193,33 @@ proc tz*(db: TimezoneDb, tzName: string): Timezone {.raises: [ValueError].} =
                 "Timezone does not exist in database: " & tzName)
         result = newTimezone(tz)
 
+proc setDefaultTzDb*(db: TimezoneDb) =
+    ## Sets the timezone database that will be used for ``tz(tzName)`` and
+    ## ``tzInfo(tzName)``. The default timezone database is stored in a thread
+    ## local variable, so calling this only affects the calling thread!
+    defaultTzDb = db
+
+proc getDefaultTzDb*(): TimezoneDb =
+    ## Gets the timezone database that is used for ``tz(tzName)`` and
+    ## ``tzInfo(tzName)``. The default timezone database is stored in a thread
+    ## local varaible, so calling ``setDefaultTzDb`` only affects the calling
+    ## thread!
+    try:
+        if defaultTzDb.empty:
+            # thread-level initialization. static content is still used
+            setDefaultTzDb(parseTzDb(content))
+    except Exception:
+        raise newException(ValueError, "can not find local tz db")
```

Comment:
This is to play nice with the effect system to only `{.raises: [ValueError].} ` instead of `IOError, OSError` and `Exception`. As an alternative, we can let this pass through and modify the necessary `.raises`?

Reply:
Lets just remove the affected `{.raises.}` declarations, the `json` module (which parseTzDb uses) doesn't play nice with exception tracking